crypto/src/main/scala/coop/rchain/crypto/util/CertificateHelper.scala

```diff
@@ -110,14 +118,49 @@ object CertificateHelper {
     val cert = new X509CertImpl(info)
     cert.sign(privateKey, algorythm)
 
-    // Update the algorith, and resign.
+    // Update the algorithm, and resign.
     val algorithmId2 = cert.get(X509CertImpl.SIG_ALG).asInstanceOf[AlgorithmId]
     info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algorithmId2)
     val cert2 = new X509CertImpl(info)
     cert2.sign(privateKey, algorythm)
     cert2
   }
 
+  def encodeSignatureRStoDER(signatureRS: Array[Byte]) = {
+    def toASN1Int(bytes: Array[Byte]) = new ASN1Integer(BigIntegers.fromUnsignedByteArray(bytes))
+
+    val (r, s) = signatureRS.take(64).splitAt(32)
+    val bos    = new ByteArrayOutputStream(72)
+    val seq    = new DERSequenceGenerator(bos)
+    try {
+      seq.addObject(toASN1Int(r))
+      seq.addObject(toASN1Int(s))
+      seq.close
+      bos.toByteArray
+    } catch { case _: Throwable => Array[Byte]() } finally {
+      if (seq != null) seq.close()
```

Comment:
This can throw an exception I think, which means that `bos` will never get closed

Reply:
It seems that closing `bos` is not necessary, but exception to close `seq` must be handled.
> Closing a ByteArrayOutputStream has no effect. The methods in this class can be called after the stream has been closed without generating an IOException.
https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html